### PR TITLE
scootapi/setup: improve waiting and killing

### PR DESCRIPTION
**Pull request by dbentley**
Problem: if you start a command with Cmds, and it then starts another command, Cmds.Kill() will not kill it. This is especially problematic if the command it starts is a server that binds to a port.

Solution: Cmds.Kill now sends the signal to the entire process group. It also starts processes in a new process group.

(We weren't hitting this in open source, but were internally, where our saga log implementation is a server on a port)

### CODE REVIEW

**Comment by dbentley** [comment]
I noticed that some (internal) commands weren't nicely dying, because they were spawning processes. This switches us to sending signals to process groups instead of processes.

**Comment by dbentley** [comment]
Ugh, nope, this won't work with this approach. NM.

**Comment by jasonbs10** [comment]
Can you remind me how user commands like 'smoketest' exit cleanly? I can't seem to find any killing of the 'setup' stuff after a clean cmd exit...

What was broken about this?

**Comment by dbentley** [comment]
Updated the PR description; how's that?

File: scootapi/setup/cmds.go [hunk 1]
```diff
@@ -44,26 +48,70 @@ func NewSignalHandlingCmds(tmp *temp.TempDir) *Cmds {
...
+	}
+	c.mu.Lock()
+
+	if len(c.watching) == 0 {
+		return
+	}
+
+	time.Sleep(10 * time.Second)
+
+	// First, send SIGINT to each
```
Comment (dgassaway) [inline scootapi/setup/cmds.go]: comment - SIGKILL

File: scootapi/setup/cmds.go [hunk 2]
```diff
@@ -44,26 +48,70 @@ func NewSignalHandlingCmds(tmp *temp.TempDir) *Cmds {
...
+	case <-time.After(5 * time.Second):
+		log.Printf("Still waiting; killing all")
+	}
+	c.mu.Lock()
+
+	if len(c.watching) == 0 {
+		return
+	}
+
+	time.Sleep(10 * time.Second)
```
Comment (dgassaway) [inline scootapi/setup/cmds.go]: Shouldn't need to sleep here again? Or change time.After() in select above to reflect actual patience?

Comment (dbentley) [inline scootapi/setup/cmds.go]: Thx; fixed.

**Pull request merged**